Movement.cpp

```diff
@@ -1,80 +1,86 @@
-/* Roborregos 2020.
+/* ROBORREGOS MAZE 2020.
  * This Movement class has all the functions
  * To move the robot to anywhere on the map.
  * This class works with all robot enginees.
  * To get more information, go to Movement.h file.
 */
 #include "Movement.h"
 
-Movement::Movement(){}
+Movement::Movement() {}
 
-Movement::Movement(BNO *b, Control *s, Motors *r) {
-    bno_ = b;
-    system_ = s;
-    robot_ = r;
+Movement::Movement(BNO *bno, Control *system, Motors *robot)
+{
+  bno_ = bno;
+  system_ = system;
+  robot_ = robot;
 }
 
-void Movement::advancePID(const double desire) {
+void Movement::advancePID(const double desire)
+{
   encoder_count_left = 0;
   encoder_count_right = 0;
 
-  do {
-      double pwm_right_enginees = system_->getPwmBNORight(desire); // Negative.
-      double pwm_left_enginees = system_->getPwmBNOLeft(desire); // Positive. 
-      if (pwm_left_enginees > 1) {
-          pwm_right_enginees = Common::kLimit_inf_pwm;
-          system_->getPwm(pwm_left_enginees);
-      }
-
-      else if (pwm_right_enginees < 0) {
-          pwm_left_enginees = Common::kLimit_inf_pwm;
-          system_->getPwm(pwm_right_enginees);
-      }
-      robot_->forwardPwm(pwm_right_enginees, pwm_left_enginees);
+  do
+  {
+    double pwm_right_enginees = system_->getPwmBNORight(desire) + system_->getPwmUltrasonicRight(); // Negative.
+    double pwm_left_enginees = system_->getPwmBNOLeft(desire) + system_->getPwmUltrasonicLeft();    // Positive.
```

Comment:
This is nice, simply double check if the math is correct What I mean by this is: Are you sure adding up the values is the correct way? Should you average them? What are the range of possible values that it can give you?